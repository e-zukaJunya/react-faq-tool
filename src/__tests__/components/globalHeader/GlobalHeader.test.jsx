import GlobalHeader from "components/globalHeader/GlobalHeader";
import { mount, shallow } from "enzyme";
import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";

// 子コンポーネントのモック
jest.mock("components/globalHeader/AccountMenu", () => () => <div>mockAccountMenuCom</div>);
jest.mock("components/globalHeader/HoverMenu", () => () => (
  <div className={"mockHoverMenuCom"}>mockHoverMenuCom</div>
));

// jsdomにcreateRangeを実装
document.createRange = () => ({
  setStart: () => {},
  setEnd: () => {},
  commonAncestorContainer: {
    nodeName: "BODY",
    ownerDocument: document
  }
});

describe("AlertDialog", () => {
  const tabs = [
    { category: "ダッシュボード", pages: [{ id: "", name: "ダッシュボード" }] },
    {
      category: "KB",
      pages: [
        { id: "kbAdd", name: "追加" },
        { id: "kbEdit", name: "編集" },
        { id: "kbApply", name: "適用" }
      ]
    }
  ];
  const getMaster = jest.fn();

  test("mount", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/"]}>
        <GlobalHeader tabs={tabs} getMaster={getMaster} />
      </MemoryRouter>
    );
    const dom = wrapper.getDOMNode();
    expect(dom).toMatchSnapshot();

    const mockComs = wrapper.find(".mockHoverMenuCom");
    expect(mockComs.length).toBe(tabs.length);
  });

  test("click button", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/"]}>
        <GlobalHeader tabs={tabs} getMaster={getMaster} />
      </MemoryRouter>
    );
    const button = wrapper.find("button");
    button.simulate("click");
    expect(getMaster).toHaveBeenCalled();
  });
});
