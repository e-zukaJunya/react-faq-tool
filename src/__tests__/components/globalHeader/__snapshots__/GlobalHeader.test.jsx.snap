// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertDialog mount 1`] = `
<div
  class=""
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="menu"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        管理ツール
      </h6>
      <div
        class="mockHoverMenuCom"
      >
        mockHoverMenuCom
      </div>
      <div
        class="mockHoverMenuCom"
      >
        mockHoverMenuCom
      </div>
      <div>
        mockAccountMenuCom
      </div>
    </div>
  </header>
</div>
`;
